### PR TITLE
 intial logging mechanism and deactive account

### PR DESCRIPTION
**Pull request by Rupeshiya**
---
name: Pull Request template
about: Describe the PR.
title: ''
labels: ''
assignees: ''

---

### **Problem**
Github Issue Number: #155 
As of now, it's pretty tough to debug using the console because there is no proper logging mechanism._

### **Solution of problem**
Add some logging mechanisms to debug in a development or production environment.

### **Type of Change**
[ ] Bug fix
[&check;] New Feature
[ ] Development of UI/UX prototypes
[&check;] Small refactor
[ ] Change in Documentation

### **Checklist**
[&check;] My code follows the same style as the codebase
[ ] My Code change requires a change in documentation
[ ] I have updated the Readme accordingly
[&check;] I made PR against **development branch**
[ ] I have run the test cases locally and it's passing.
[&check;] I have squashed my commits

### CODE REVIEW

**Comment by Rupeshiya** [comment]
@devesh-verma Please review

File: config/winston.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    new winston.transports.Console(options.console)
+  ],
+  exitOnError: false // do not exit on handled exceptions
+})
+
+// create a stream object with a 'write' function that will be used by `morgan`
+logger.stream = {
+  write: function (message, encoding) {
+    // use the 'info' log level so the output will be picked up by both transports (file and console)
+    logger.info(message)
```
Comment (devesh-verma) [inline config/winston.js:60]: @Rupeshiya by doing this we will be logging info in console even if we dont want to.
Can we modify this such that in development environment we prints logs on browser console as well as server side logs and in prod env we do it just on the server side.

File: app/controllers/auth.js [hunk 2]
```diff
@@ -9,9 +9,6 @@ module.exports = {
       const token = await user.generateAuthToken()
       res.send({ user: user, token: token })
     } catch (error) {
-      if (process.env.NODE_ENV !== 'production') {
-        console.log(error.name, '-', error.message)
-      }
```
Comment (devesh-verma) [inline app/controllers/auth.js:14]: if you are removing these logs how would we have a log when something goes wrong? and what was the data as this particular time when the error occoured?

File: config/winston.js (hunk 1 above)
Comment (Rupeshiya) [reply config/winston.js:60]: Yes, @devesh-verma I have already done that part for production in frontend part like in prod there would not be anything in the browser console.

And If we want to have log both in development and production in server-side then why we need to check env??

File: app/controllers/auth.js (hunk 2 above)
Comment (Rupeshiya) [reply app/controllers/auth.js:14]: yes, you could check the logs in root directory of the folder /app/logs/app.log. There you could find all those logs where error occured and why occured etc.

File: config/winston.js (hunk 1 above)
Comment (Rupeshiya) [reply config/winston.js:60]: These are not affecting browser console in anyway.

File: app/controllers/auth.js (hunk 2 above)
Comment (Rupeshiya) [reply app/controllers/auth.js:14]: If you want to check it, just check this PR locally once

File: config/winston.js (hunk 1 above)
Comment (devesh-verma) [reply config/winston.js:60]: not on server-side, I was talking with respect to client-side. This is good then 👍

File: app/controllers/auth.js (hunk 2 above)
Comment (devesh-verma) [reply app/controllers/auth.js:14]: perfect 👍

Comment (Rupeshiya) [reply app/controllers/auth.js:14]: See the log: The bold part is the data 
```
[2020-07-26 14:52:09] info: ::1 - - [26/Jul/2020:09:22:09 +0000] "GET /user/overview" 304 - "http://localhost:3000/dashboard" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.38 Safari/537.36" **{}**
```

**Pull request merged**